src/extensions/default/bramble-move-file/MoveToDialog.js

```diff
@@ -42,6 +49,18 @@ define(function (require, exports, module) {
             // in the folder that is being moved.
             if(fileInEditor.indexOf(source) === 0) {
                 CommandManager.execute(Commands.CMD_OPEN, {fullPath: fileToOpen});
+                // Once the file tree has been refreshed, a `change` event
+                // on the FileSystem is fired.
+                // This is unfortunately the only way to make sure
+                // the file tree first reflects the (delete + create) move
+                // operation and only then do `afterFileTreeRefresh` which
+                // expands the parent directories and reloads the preview.
+                // If it is not done in this order, you'll see the directories
+                // expand first with the file shown in its old location and
+                // after a delay, the file will jump to the new location.
+                FileSystem.on("change", afterFileTreeRefresh);
```

Comment:
Why not just do `.one` instead of `on` so you don't have to remove this listener above?  It's either `one` or `once`, I can't remember which they use in Brackets.